collector/app.go

```diff
@@ -32,3 +105,64 @@ func GetAppsCollection(c *CollectorOpts, url string) *client.AppCollection {
 
 	return appCollection
 }
+
+func GetAllAppData(c *CollectorOpts, url string) []rancher.App {
+	appList := GetAppCollection(c, url)
+	if appList == nil {
+		return nil
+	}
+
+	appData := appList.Data
+	// Paginating if needed
+	appList, err := appList.Next()
+	for page := 2; appList != nil && err == nil; appList, err = appList.Next() {
+		log.Debug(" Paginating Applications page ", page)
+		appData = append(appData, appList.Data...)
+		page++
+	}
+	if err != nil {
+		log.Errorf(" Failed to paginate Applications err=%s", err)
+		return nil
+	}
+
+	return appData
+}
+
+func SplitAppExternalID(externalID string) map[string]string {
```

Comment:
Instead of splitting the string and looping, use `url.Parse` and check the query keys, [example](https://github.com/rancher/rancher/blob/master/pkg/controllers/user/helm/common/common.go#L63). I would try to copy logic in rancher as much as possible here. 
Also can you add a comment explaining what this returns
Same for MCA method.

Reply:
Updated